@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -430,62 +422,63 @@ function placePointLabelChoosingAnchor(
     // Store label state - persistent or new label.
     const persistent = labelState.visible;
 
-    // Text elements with multi-anchor placement require their own layout style.
-    // Provide private copy of label layout and attach it to current label and text canvas.
-    label.layoutStyle = label.layoutStyle!.clone();
-    textCanvas.textLayoutStyle = label.layoutStyle!;
-
     // The current implementation does not provide placements options via theme style yet,
     // so function tries the anchor placements from pre-defined placements arrays.
     // TODO: HARP-6487 Placements options should be loaded from the theme.
 
-    // Start with currently set alignment settings.
-    const basePlacement = {
-        h: label.layoutStyle!.horizontalAlignment,
-        v: label.layoutStyle!.verticalAlignment
-    };
+    // Start with recently set alignment settings if layout state is stored or
+    // simply begin from layout defined in theme.
+    const labelThemeLayout = label.layoutStyle!;
+    const labelStateLayout = labelState.textLayoutState;
+    const recentPlacement =
```

Comment:
Better name: 'lastPlacement'. Shorter, and more precise.

Reply:
Ok.